infrastructure/shared-vpc/main.tf

```diff
@@ -0,0 +1,187 @@
+  routes           = []
+}
+
+# Shared VPC firewall
+
+module "net-vpc-firewall" {
+  # source               = "terraform-google-modules/network/google//modules/fabric-net-firewall"
+  # version              = "1.4.0"
+  source               = "github.com/terraform-google-modules/terraform-google-network//modules/fabric-net-firewall?ref=52e49a9"
+  project_id           = module.project-svpc-host.project_id
+  network              = module.net-vpc-host.network_name
+  admin_ranges_enabled = true
+  admin_ranges         = [lookup(local.net_subnet_ips, "networking")]
+  custom_rules = {
+    ingress-mysql = {
+      description          = "Allow incoming connections on the MySQL port from GKE addresses."
+      direction            = "INGRESS"
+      action               = "allow"
+      ranges               = local.net_gke_ip_ranges
+      sources              = []
+      targets              = ["mysql"]
+      use_service_accounts = false
+      rules                = [{ protocol = "tcp", ports = [3306] }]
+      extra_attributes     = {}
+    }
+  }
+}
+
+# Shared VPC access
+
+module "network_fabric-net-svpc-access" {
+  source              = "terraform-google-modules/network/google//modules/fabric-net-svpc-access"
+  version             = "1.3.0"
+  host_project_id     = module.project-svpc-host.project_id
+  service_project_num = 2
+  service_project_ids = [
+    module.project-service-gce.number, module.project-service-gke.number
+  ]
+  host_subnets = ["gce", "gke"]
+  host_subnet_regions = [
+    local.net_subnet_regions["gce"], local.net_subnet_regions["gke"]
+  ]
+  host_subnet_users = {
+    gce = join(",", local.net_gce_users)
+    gke = join(",", local.net_gke_users)
+  }
+  host_service_agent_role = true
+  host_service_agent_users = [
+    "serviceAccount:${module.project-service-gke.gke_service_account}"
+  ]
+}
+
+################################################################################
+#                                     DNS                                      #
+################################################################################
+
+module "host-dns" {
+  source                             = "terraform-google-modules/cloud-dns/google"
+  version                            = "2.0.0"
+  project_id                         = module.project-svpc-host.project_id
+  type                               = "private"
+  name                               = "svpc-fabric-example"
+  domain                             = "svpc.fabric."
+  private_visibility_config_networks = [module.net-vpc-host.network_self_link]
+  record_names                       = ["localhost"]
+  record_data                        = [{ rrdatas = "127.0.0.1", type = "A" }]
+}
+
+################################################################################
+#                                     KMS                                      #
+################################################################################
+
+module "host-kms" {
```

Comment:
When I'm trying to destroy the example:
```
Error: Instance cannot be destroyed

  on .terraform/modules/host-kms/terraform-google-modules-terraform-google-kms-4237838/main.tf line 27:
  27: resource "google_kms_crypto_key" "key" {

Resource module.host-kms.google_kms_crypto_key.key[0] has
lifecycle.prevent_destroy set, but the plan calls for this resource to be
destroyed. To avoid this error and continue with the plan, either disable
lifecycle.prevent_destroy or reduce the scope of the plan using the -target
flag.
```
I can see that `google-kms` module does not have it as an option, the questions are:
1. how to destroy?
2. Should we have some notes in readme about that?

Reply:
Thanks for catching this. I'll submit a PR to the KMS module to add support for setting `prevent_destroy` on keys. I seem to remember we have the same variable in the GCS module for buckets.